src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs

```diff
@@ -425,6 +425,8 @@ public JsonObjectSerialization BuildJsonObjectSerialization(InputModelType input
             PopulatePropertyBag(propertyBag, 0);
             PopulatePropertyBag(selfPropertyBag, 0);
 
+            // properties: all the properties containing the properties from base needed to build the constructor
+            // selfProperties: the properties not containing properties from base just used to do the serialization
             var properties = GetPropertySerializationsFromBag(propertyBag, objectType).ToArray();
```

Comment:
I am more curious why we need both.
If this is a temporary solution, please add TODO and/or open an issue to track.
If this is permanent, I think we need comments to explain why we need both.
The serialization objects might not have the hierarchy and we have to do this, this is an acceptable explanation, but we need to write them somewhere so that in the future we will not be confused.

Reply:
Because from properties we don't know which is from base, which is from itself. Not sure if it is what you said "The serialization objects might not have the hierarchy".

Reply:
I updated the description to make it more clear.

Reply:
I get the purpose now.
So I regard this as a temporary solution, we have to keep two list of properties because we do not know which property comes from myself and which comes from my base.
This leads to an immediate conclusion that the serialization should have hierarchy.
The serialization object on myself should only have my properties, and it has a reference like `Base` just like what we have in types.
And now if the flag of using write core is turned off, we could go through all the bases to collect all the properties, and if it is turned on, I believe the only thing we would care when writing this model is "my properties".
This should cover both scenarios.

But this change of implementation could be complicated to do right now. We should track it with an issue and refactor it in the future if possible, or at least keep in mind when we write the azure plugin, we should not do the things in this PR again.

Reply:
I created an issue https://github.com/Azure/autorest.csharp/issues/5024